src/gui/Gui.cpp

```diff
@@ -3,84 +3,313 @@
-   void OnExit(wxCommandEvent& event);
-   void OnAbout(wxCommandEvent& event);
-   static wxSize FixedSize;
+   // Here you put the frame functions:
+   bool DarkMode = wxSystemSettings::GetAppearance().IsDark();
 };
+
+
 enum { ID_Hello = 1 };
-wxSize MyFrame::FixedSize(370, 400);
 
+
+/////////// Event Table ///////////
+wxBEGIN_EVENT_TABLE(MyMainFrame, wxFrame)
+   EVT_BUTTON(39, MyMainFrame::OnClickAccount)
+   EVT_BUTTON(40, MyMainFrame::OnClickSettings)
+   EVT_BUTTON(41, MyMainFrame::OnClickLaunch)
+   EVT_BUTTON(42, MyMainFrame::OnClickLogo)
+wxEND_EVENT_TABLE()
+
+
+
+/////////// OnInit function to show frame ///////////
 bool MyApp::OnInit() {
-   auto* frame = new MyFrame();
-   frame->Show(true);
+   auto* MainFrame = new MyMainFrame();
+   MainFrame->SetIcon(wxIcon("BeamMP.png",wxBITMAP_TYPE_PNG));
+
+   // Set MainFrame properties:
+   MainFrame->SetSize(1000, 650);
+   MainFrame->Center();
+
+   if (wxSystemSettings::GetAppearance().IsDark()) {
+      MainFrame->SetBackgroundColour(wxColour(40, 40, 40));
+      MainFrame->SetForegroundColour(wxColour(255, 255, 255));
+   }
+
+   else {
+      MainFrame->SetBackgroundColour(wxColour("white"));
+      MainFrame->SetForegroundColour(wxColour("white"));
+   }
+
+   wxFileSystem::AddHandler(new wxInternetFSHandler);
+   MainFrame->Show(true);
    return true;
 }
 
-MyFrame::MyFrame() :
-    wxFrame(nullptr, wxID_ANY, "BeamMP V3.0", wxDefaultPosition, FixedSize) {
-   // SetMaxSize(FixedSize);
-   // SetMinSize(FixedSize);
-   Center();
-
-   // 27 35 35
-
-   wxColour Colour(27, 35, 35, 1);
-   SetBackgroundColour(Colour);
-
-   auto* menuFile = new wxMenu;
-   menuFile->Append(ID_Hello, "&Hello...\tCtrl-H",
-                    "Help string shown in status bar for this menu item");
-   menuFile->AppendSeparator();
-   menuFile->Append(wxID_EXIT);
-   auto* menuHelp = new wxMenu;
-   menuHelp->Append(wxID_ABOUT);
-   auto* menuBar = new wxMenuBar;
-
-   menuBar->SetOwnBackgroundColour(Colour);
-   menuBar->Append(menuFile, "&File");
-   menuBar->Append(menuHelp, "&Help");
-   // SetMenuBar(menuBar);
-
-   auto* m_ani = new wxAnimationCtrl(this, wxID_ANY);
-   wxMemoryInputStream stream(gif::Logo, sizeof(gif::Logo));
-   if (m_ani->Load(stream)) m_ani->Play();
-
-   Bind(wxEVT_MENU, &MyFrame::OnHello, this, ID_Hello);
-   Bind(wxEVT_MENU, &MyFrame::OnAbout, this, wxID_ABOUT);
-   Bind(wxEVT_MENU, &MyFrame::OnExit, this, wxID_EXIT);
+/////////// MainFrame Function ///////////
+MyMainFrame::MyMainFrame() :
+    wxFrame(nullptr, wxID_ANY, "BeamMP Launcher V3", wxDefaultPosition,wxDefaultSize,
+            wxMINIMIZE_BOX | wxSYSTEM_MENU | wxCAPTION | wxCLOSE_BOX) {
+
+   /////////// News ///////////
+   wxWebView::New() ->Create(this, wxID_ANY, "https://beammp.com", wxPoint(10, 70), wxSize(950, 400));
+   auto* txtNews = new wxStaticText(this, wxID_ANY, wxT("News"), wxPoint(10, 40));
+
+   auto* HorizontalLine1 = new wxStaticLine(this, wxID_ANY, wxPoint(10, 60), wxSize(950, 1));
```

Comment:
I can't test this, but is this resizable? Like, if these are pixel values, does wx know how to constrain them when resizing the window?

Reply:
The window cannot be resized in any way. It's a launcher like other game launchers (cannot be maximized, cannot be resized...)

Reply:
What's the reasoning behind this?

Reply:
That's how launchers are made, for example: Nitrox Launcher, League of Legends Launcher...

Reply:
In that case (no technical limitation) we should probably ask our users. Generally a few polls regarding UI would be good, as we have a few UI/UX experienced people on the team, and in the community.

Reply:
But why, they can get used to it not being fullscreen and like the games I mentioned above. You can't satisfy everyone. Also, it is easier for us to develop it having fixed dimensions because we are adding the options directly on the window with specified positions. Resizing the window might cause issues with the current code.
Suggestion: Keep it fixed for now, we finish the entire thing, and later on with another version we can update everything. Can't perfect work from version 1!

Reply:
I was just being nice, we're definitely polling people on their opinions on design decisions within the UI. Tens of thousands of people have to use this, with different screen layouts, screen sizes, preferences, etc., every UI/UX decision should be considered carefully. That's why - I'm not even sure why "let's ask the users" would lead to a "but why".

Reply:
Of course keep it fixed for now, but for the final version, we need to ask users and see what they'd like, just like we should do for most aspects of UI

Reply:
Being nice is fine to a certain point. The users will have different opinions about the color, UI, window resizing, and all of that. It is better for us to determine how it looks at least for V1 and will accept suggestions later. Like I mentioned before, League of Legends, which is a very popular game, has a launcher with fixed dimensions. For me personally, it was a bit annoying when I started playing the game, but now it seems fine to me. It's a launcher, not the actual game, so it's not bad if it's not fullscreen. We can accept the users' feedback about this topic but we should leave the launcher like this for now until it is ready for us to update it later.